### PR TITLE
fix(demos): update React demos to React InstantSearch Hooks

### PR DESCRIPTION
**Pull request by FabienMotte**
### Description

This PR updates the demos using React InstantSearch to React InstantSearch Hooks for the 4 industries (Ecommerce, Geo, Media and SaaS).

[FX-2372](https://algolia.atlassian.net/browse/FX-2372)

#### Common

Each demo has been generated using [create-instantsearch-app](https://github.com/algolia/instantsearch/tree/master/packages/create-instantsearch-app).

- Replaced `react-scripts` by `parcel`.
- Used TypeScript instead of JavaScript.
- Replaced `react-instantsearch` by `react-instantsearch-hooks-web`.
- "Open Sans" Google font is now loaded from the `index.html` instead of the `App.css` file.
- Formatted relevant files using Prettier.

#### Media demo

- Added a couple of `searchableAttributes` in the Dashboard configuration of the index `demo_media`.
- Added `categories` as a facet since it was missing and not displaying the refinement list properly.
- Added an image placeholder for each hit since the images from the media dataset are broken for now.
[They will be fixed later on.](https://algolia.atlassian.net/browse/FX-2386)

#### SaaS demo

- Deleted `Switch` component since it's not used.

#### Geo demo

- Based on the [`geo-search` doc code sample](https://github.com/algolia/doc-code-samples/tree/master/react-instantsearch-hooks/geo-search).
- Replaced `react-instantsearch-dom-maps` by `react-leaflet`.

[FX-2372]: https://algolia.atlassian.net/browse/FX-2372?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: demos/react/ecommerce/src/App.tsx [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  Configure,
+  InstantSearch,
+  Pagination,
+  RangeInput,
+  RefinementList,
+  SearchBox,
+} from 'react-instantsearch-hooks-web'
+
+import { Content } from './Content'
+import { Stats } from './Stats'
```
Comment (Haroenv) [inline demos/react/ecommerce/src/App.tsx:13]: stats exists in React InstantSearch Hooks

File: demos/react/ecommerce/src/Stats.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import React from 'react'
+import { useInstantSearch } from 'react-instantsearch-hooks-web'
+
+export function Stats() {
+  const { results } = useInstantSearch()
```
Comment (Haroenv) [inline demos/react/ecommerce/src/Stats.tsx:5]: unless we specifically teach custom widgets, the basic Stats widget is probably sufficient right?

File: demos/react/geo/src/Airports.tsx [hunk 3]
```diff
@@ -0,0 +1,92 @@
+import React, { useState } from 'react'
+import { Marker, Popup, useMapEvents } from 'react-leaflet'
+import { DivIcon } from 'leaflet'
+import { useSearchBox } from 'react-instantsearch-hooks-web'
+import { useGeoSearch } from './useGeoSearch'
```
Comment (Haroenv) [inline demos/react/geo/src/Airports.tsx:5]: unrelated to the PR, but we probably should at least expose this one in React InstantSearch Hooks too

**Review by Haroenv — COMMENTED** [review]
👍 looks good!

File: demos/react/ecommerce/src/App.tsx (hunk 1 above)
Comment (FabienMotte) [reply demos/react/ecommerce/src/App.tsx:13]: Oh I didn't know that, we only document the [hook `useStats`](https://www.algolia.com/doc/api-reference/widgets/stats/react-hooks/) for now, I think we should document the widget instead, WDYT?

Comment (Haroenv) [reply demos/react/ecommerce/src/App.tsx]: yep, we should document the widget too I think

**Review by Haroenv — APPROVED** [review]

File: demos/react/ecommerce/src/App.tsx (hunk 1 above)
Comment (FabienMotte) [reply demos/react/ecommerce/src/App.tsx]: I'll create two tickets for RISH, one to document the `Stats` widget, and one to expose the `useGeoSearch` hook.

**Pull request merged**